source/credentials.c

```diff
@@ -213,34 +230,82 @@ struct aws_credentials_provider *aws_credentials_provider_new_chain_default(
         providers[index++] = profile_provider;
     }
 
-    struct aws_credentials_provider_imds_options imds_options;
-    AWS_ZERO_STRUCT(imds_options);
-    imds_options.bootstrap = options->bootstrap;
-    imds_provider = aws_credentials_provider_new_imds(allocator, &imds_options);
-    if (imds_provider == NULL) {
-        goto on_error;
-    }
+    if (ecs_relative_uri && ecs_relative_uri->len) {
+        struct aws_credentials_provider_ecs_options ecs_options;
+        AWS_ZERO_STRUCT(ecs_options);
+        ecs_options.bootstrap = options->bootstrap;
+        ecs_options.host = aws_byte_cursor_from_string(s_ecs_host);
+        ecs_options.path_and_query = aws_byte_cursor_from_string(ecs_relative_uri);
+        ecs_options.use_tls = false;
+        ecs_provider = aws_credentials_provider_new_ecs(allocator, &ecs_options);
+        if (ecs_provider == NULL) {
+            success = false;
+            goto on_ret;
+        }
+        providers[index] = ecs_provider;
```

Comment:
why isn't this `providers[index++] = ecs_provider` ?

Reply:
I had the same question and think your idea is the correct way to do, but not quit sure why was this way originally. (originally the first two are p[index++] = provider, but the last one didn't increase index)

Reply:
Talked with Jonathan who put this index variable in to the function. He agrees we should just be `++`ing it everywhere